File: player/py/player_service.py
#!/usr/bin/env python
 
import sys
sys.path.append('./lib/api')
  
from player_strategy import PlayerStrategy
from player_strategy.ttypes import *
   
from thrift.transport import TSocket
from thrift.transport import TTransport
from thrift.protocol import TBinaryProtocol
from thrift.server import TServer
    
import socket

from lib.handler import PlayerHandler

handler = PlayerHandler()
processor = PlayerStrategy.Processor(handler)
transport = TSocket.TServerSocket('localhost', 30303)
tfactory = TTransport.TBufferedTransportFactory()
pfactory = TBinaryProtocol.TBinaryProtocolFactory()

server = TServer.TSimpleServer(processor, transport, tfactory, pfactory)

print "Starting python server..."
server.serve()
print "done!"

File: player/php/player_service.php
#!/usr/bin/env php
<?php

error_reporting(E_ALL);

require_once __DIR__.'/lib/api/Thrift/ClassLoader/ThriftClassLoader.php';
require_once 'lib/handler.php';

use Thrift\ClassLoader\ThriftClassLoader;

$GEN_DIR = realpath(dirname(__FILE__)).'/lib/api';

$loader = new ThriftClassLoader();
$loader->registerNamespace('Thrift', __DIR__ . '/lib/api');
$loader->registerDefinition('shared', $GEN_DIR);
$loader->registerDefinition('API', $GEN_DIR);
$loader->register();


if (php_sapi_name() == 'cli') {
  ini_set("display_errors", "stderr");
}

use Thrift\Protocol\TBinaryProtocol;
use Thrift\Transport\TPhpStream;
use Thrift\Transport\TBufferedTransport;

header('Content-Type', 'application/x-thrift');
if (php_sapi_name() == 'cli') {
  echo "\r\n";
}

$handler = new \PlayerHandler();
$processor = new \API\PlayerStrategyProcessor($handler);

$transport = new TBufferedTransport(new TPhpStream(TPhpStream::MODE_R | TPhpStream::MODE_W));
$protocol = new TBinaryProtocol($transport, true, true);

$transport->open();
$processor->process($protocol, $protocol);
$transport->close();